### PR TITLE
Ensure the directory of the zip entry exists before running realpath

### LINKED ISSUE #38: Regression between 1.6.0 and 1.6.1 with directories in zips
We've noticed a regression in `extract-zip` when trying to `npm install electron`.

[This line](https://github.com/electron-userland/electron-prebuilt/blob/bff34212b7a720ffc4ea5c6410ee08219934030f/install.js#L38) fails with this error (ran the file manually, to see which version caused the regression):

```
Error: ENOENT: no such file or directory, lstat '/tmp/electron-prebuilt/dist/resources'
```

When the script fails, I noticed that neither folder (`resources` or `locales`) were in the extracted folder. My guess is that it's not recursively creating subfolders correctly.

See also: https://github.com/electron/electron/issues/9323

### PR DESCRIPTION
**Pull request by MarshallOfSound**
`realpath` requires that the directory path it is attempting to resolve actually exists, so we need to create it before running the check as not all zip files create the directory before extracting a file into it.

This fixes the current issue with `electron-prebuilt`

/cc @malept @maxogden 

Fixes #38.

### CODE REVIEW

**Comment by malept** [comment]
I don't suppose you can fix the tests? (And possibly add a test with a directory in it?)

**Comment by MarshallOfSound** [comment]
They pass on my machine, don't really have time to look at them 😢 

Fork is open to contribs to push to though if someone wants to pick this up

**Comment by malept** [comment]
> Fork is open to contribs to push to though if someone wants to pick this up

Done :smile:

File: test/test.js [hunk 1]
```diff
@@ -1,119 +1,112 @@
...
-
-var sourceE = path.join(__dirname, 'symlink-dest.zip')
-var targetE = os.tmpdir()
-var resultsE = path.join(targetB, 'symlink-dest')
-
-test('extract cat zip', function (t) {
-  rimraf.sync(targetA)
+function mkdtemp (suffix, callback) {
+  temp.mkdir({prefix: 'extract-zip', suffix: suffix}, function (err, dirPath) {
+    if (err) throw err
```
Comment (MarshallOfSound) [inline test/test.js]: This should probably be `t.same(err, null)` the throw will be lost in a long callback chain

Comment (malept) [inline test/test.js]: I went with `t.notOk`.

**Comment by HoWingYip** [comment]
There was a workaround to set "extract-zip" to "=1.6.0", but I still get the same error when it's running "node install.js". I'm trying to install the Electron framework via npm 4.6.1. I have reinstalled Node.js and therefore npm. This workaround does not seem to work.

**Comment by malept** [comment]
Please be patient, I'm in the middle of releasing 1.6.5 with this fix.

**Comment by HoWingYip** [comment]
I am sorry.

**Comment by HoWingYip** [comment]
FIXED! Thank you so much @malept!

**Comment by malept** [comment]
Be sure to thank @MarshallOfSound too, he's the one who wrote the actual fix.

**Comment by HoWingYip** [comment]
Thanks, @MarshallOfSound!

**Pull request merged**